src/Config/ConfigOverrides.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace Drupal\stanford_profile_helper\Config;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ConfigFactoryOverrideInterface;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\State\StateInterface;
+use Drupal\config_pages\ConfigPagesLoaderServiceInterface;
+use Drupal\config_pages\ConfigPagesInterface;
+use Drupal\Core\StreamWrapper\StreamWrapperManagerInterface;
+
+/**
+ * Config overrides for stanford profile.
+ *
+ * @package Drupal\stanford_profile_helper\Config
+ */
+class ConfigOverrides implements ConfigFactoryOverrideInterface {
+
+  /**
+   * State service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * Config pages loader service.
+   *
+   * @var \Drupal\config_pages\ConfigPagesLoaderServiceInterface
+   */
+  protected $configPagesLoader;
+
+  /**
+   * Config factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Config Entity Type Manager Interface.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * StreamWrapperInterface service.
+   *
+   * @var \Drupal\Core\StreamWrapper\StreamWrapperManagerInterface
+   */
+  protected $streamWrapperManager;
+
+  /**
+   * ConfigOverrides constructor.
+   *
+   * @param \Drupal\Core\State\StateInterface $state
+   *   State service.
+   * @param \Drupal\config_pages\ConfigPagesLoaderServiceInterface|null $config_pages_loader
+   *   Config pages service.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface|null $config_factory
+   *   Config factory service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface|null $entity_type_manager
+   *   Entity type manager interface.
+   * @param \Drupal\Core\StreamWrapper\StreamWrapperManagerInterface|null $stream_wrapper_manager
+   *   Stream wrapper manager interface.
+   */
+  public function __construct(
+    StateInterface $state,
+    ConfigPagesLoaderServiceInterface $config_pages_loader = NULL,
+    ConfigFactoryInterface $config_factory = NULL,
+    EntityTypeManagerInterface $entity_type_manager = NULL,
+    StreamWrapperManagerInterface $stream_wrapper_manager = NULL
+  ) {
+    $this->state = $state;
+
+    if ($config_pages_loader) {
+      $this->configPagesLoader = $config_pages_loader;
+    }
+
+    if ($config_factory) {
+      $this->configFactory = $config_factory;
+    }
+
+    if ($entity_type_manager) {
+      $this->entityTypeManager = $entity_type_manager;
+    }
+
+    if ($stream_wrapper_manager) {
+      $this->streamWrapperManager = $stream_wrapper_manager;
+    }
```

Comment:
```suggestion
    $this->configPagesLoader = $config_pages_loader;
    $this->configFactory = $config_factory;
    $this->entityTypeManager = $entity_type_manager;
    $this->streamWrapperManager = $stream_wrapper_manager;
```

Reply:
the original `if` condition in the profile was for upgrading existing sites to the new service. It would throw an error when then service arguments were increased until cache was cleared. Since this is a whole new service, we shouldn't have those issues.